numpy/testing/_private/pytesttester.py

```diff
@@ -0,0 +1,329 @@
+
+        _warn_opts = dict(develop=(Warning,),
+                          release=())
+        if isinstance(raise_warnings, basestring):
+            raise_warnings = _warn_opts[raise_warnings]
+
+        # run tests.
+        _show_system_info()
+        print(pytest_args)
+
+        with suppress_warnings("location") as sup:
+            # Reset the warning filters to the default state,
+            # so that running the tests is more repeatable.
+            warnings.resetwarnings()
+            # Set all warnings to 'warn', this is because the default 'once'
+            # has the bad property of possibly shadowing later warnings.
+            warnings.filterwarnings('always')
+            # Force the requested warnings to raise
+            for warningtype in raise_warnings:
+                warnings.filterwarnings('error', category=warningtype)
+            # Filter out annoying import messages.
+            sup.filter(message='Not importing directory')
+            sup.filter(message="numpy.dtype size changed")
+            sup.filter(message="numpy.ufunc size changed")
+            sup.filter(category=np.ModuleDeprecationWarning)
+            # Filter out boolean '-' deprecation messages. This allows
+            # older versions of scipy to test without a flood of messages.
+            sup.filter(message=".*boolean negative.*")
+            sup.filter(message=".*boolean subtract.*")
+            # Filter out distutils cpu warnings (could be localized to
+            # distutils tests). ASV has problems with top level import,
+            # so fetch module for suppression here.
+            with warnings.catch_warnings():
+                warnings.simplefilter("always")
+                from ...distutils import cpuinfo
+            sup.filter(category=UserWarning, module=cpuinfo)
+            # See #7949: Filter out deprecation warnings due to the -3 flag to
+            # python 2
+            if sys.version_info.major == 2 and sys.py3kwarning:
+                # This is very specific, so using the fragile module filter
+                # is fine
+                import threading
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"sys\.exc_clear\(\) not supported in 3\.x",
+                    module=threading)
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"in 3\.x, __setslice__")
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"in 3\.x, __getslice__")
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"buffer\(\) not supported in 3\.x")
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"CObject type is not supported in 3\.x")
+                sup.filter(
+                    DeprecationWarning,
+                    message=r"comparing unequal types not supported in 3\.x")
```

Comment:
I'm curious why we need this at all

Reply:
May not any more. IIRC, it was because in Python 3 not all objects could be compared.

Reply:
I'm just thinking that we now test against py3, so our testcase would fall afoul of this if we did it.

I suppose we could be testing for warnings in py2-only tests.

Reply:
That and the two above were added in 82a1d81fc6.   @juliantaylor Do you recall what that was about?

Reply:
Python 2.7 run with -3 issues a lot of deprecation warnings, and we do test with that flag. After we drop 2.7 things will be easier.